### PR TITLE
Dịch "mlp_vn" - Phần 4

### LINKED ISSUE #823: Dịch "mlp_vn" - Phần 4
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_multilayer-perceptrons/mlp_vn.md#L244-L302

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by azraelzhor**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #823

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 1]
```diff
@@ -251,14 +251,15 @@ In this example, we’ll implement an MLP with two hidden layers and one output
 ### Vectorization and Minibatch
 -->
 
-### *dịch tiêu đề phía trên*
+### Vector hoá và Batch nhỏ
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
### Vector hoá và Minibatch
```
Từ `minibatch` giữ nguyên như trong Bảng thuật ngữ nghen.

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 2]
```diff
@@ -251,14 +251,15 @@ In this example, we’ll implement an MLP with two hidden layers and one output
...
 
 <!--
 As before, by the matrix $\mathbf{X}$, we denote a minibatch of inputs.
 The calculations to produce outputs from an MLP with two hidden layers can thus be expressed:
 -->
 
-*dịch đoạn phía trên*
+Giống như trước, chúng tôi dùng ma trận $\mathbf{X}$ để ký hiệu một batch nhỏ các giá trị đầu vào.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
Giống như trước, chúng ta dùng ma trận $\mathbf{X}$ để ký hiệu một minibatch các giá trị đầu vào.
```

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 3]
```diff
@@ -276,7 +277,11 @@ That means that after computing the linear portion of the layer, we can calculat
 This is true for most activation functions (the batch normalization operation will be introduced in :numref:`sec_batch_norm` is a notable exception to that rule).
 -->
 
-*dịch đoạn phía trên*
+Khác với các quy ước thông thường, chúng tôi định nghĩa phi tuyến $\sigma$ áp dụng cho các giá trị đầu vào theo từng dòng, tức là mỗi một điểm dữ liệu quan sát được tại một thời điểm.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
Khác với các quy ước thông thường, chúng ta định nghĩa phi tuyến $\sigma$ áp dụng cho các giá trị đầu vào theo từng dòng, tức mỗi điểm dữ liệu chỉ quan sát được tại một thời điểm.
```

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 4]
```diff
@@ -276,7 +277,11 @@ That means that after computing the linear portion of the layer, we can calculat
...
+Cần chú ý rằng chúng tôi cũng sử dụng quy ước này cho *softmax* để ký hiệu toán tử tính theo từng dòng.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
Cần chú ý rằng ta cũng sử dụng quy ước này cho *softmax* để ký hiệu toán tử tính theo từng dòng.
```

File: chapter_multilayer-perceptrons/mlp_vn.md (hunk 3 above)
Comment (tiepvupsu) [reply chapter_multilayer-perceptrons/mlp_vn.md]: Mình thêm từ `hàm` vào trước `phi tuyến` trong suggestion của @duythanhvn.

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 5]
```diff
@@ -289,15 +294,17 @@ npx.set_np()
 ## Activation Functions
 -->
 
-## *dịch tiêu đề phía trên*
+## Các hàm kích hoạt
```
Comment (tiepvupsu) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
## Các hàm Kích hoạt
```

**Comment by tiepvupsu** [comment]
@azraelzhor bạn có tiếp tục hoàn thiện PR này không?

File: chapter_multilayer-perceptrons/mlp_vn.md (hunk 3 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/mlp_vn.md]: Em thắc mắc chỗ này xíu
Đoạn văn gốc là "apply to its inputs in a row-wise fashion, i.e., one observation at a time"
Theo em observation là dữ liệu quan sát được, nên nếu dịch thành "mỗi điểm dữ liệu chỉ quan sát được tại một thời điểm" có vẻ không đúng lắm?
"chỉ mỗi điểm dữ liệu quan sát được tại một thời điểm" nghe có vẻ đúng hơn?

Comment (minhduc0711) [reply chapter_multilayer-perceptrons/mlp_vn.md]: mình sẽ dịch như thế này
```suggestion
Khác với các quy ước thông thường, chúng ta định nghĩa hàm phi tuyến $\sigma$ áp dụng cho các giá trị đầu vào theo từng hàng, tức lần lượt từng điểm dữ liệu một.
```

File: chapter_multilayer-perceptrons/mlp_vn.md (hunk 1 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/mlp_vn.md]: ok

File: chapter_multilayer-perceptrons/mlp_vn.md (hunk 2 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/mlp_vn.md]: ok

File: chapter_multilayer-perceptrons/mlp_vn.md (hunk 3 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/mlp_vn.md]: ok

File: chapter_multilayer-perceptrons/mlp_vn.md (hunk 4 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/mlp_vn.md]: ok

File: chapter_multilayer-perceptrons/mlp_vn.md (hunk 5 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/mlp_vn.md]: ok

**Comment by azraelzhor** [comment]
@duythanhvn pr của mình bị failing check là sao vậy?

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 6]
```diff
@@ -276,7 +277,11 @@ That means that after computing the linear portion of the layer, we can calculat
 This is true for most activation functions (the batch normalization operation will be introduced in :numref:`sec_batch_norm` is a notable exception to that rule).
 -->
 
-*dịch đoạn phía trên*
+Khác với các quy ước thông thường, chúng ta định nghĩa hàm phi tuyến $\sigma$ áp dụng cho các giá trị đầu vào theo từng hàng, tức lần lượt từng điểm dữ liệu một.
+Cần chú ý rằng ta cũng sử dụng quy ước này cho *softmax* để ký hiệu toán tử tính theo từng dòng.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
Cần chú ý rằng ta cũng sử dụng quy ước này cho *softmax* để ký hiệu toán tử tính theo từng hàng.
```

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 7]
```diff
@@ -276,7 +277,11 @@ That means that after computing the linear portion of the layer, we can calculat
...
+Thông thường, như trong mục này, các hàm kích hoạt mà chúng ta áp dụng vào tầng ẩn không chỉ đơn thuần theo từng dòng mà còn theo từng phần.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
Thông thường, như trong mục này, các hàm kích hoạt không chỉ đơn thuần được áp dụng vào tầng ẩn theo từng hàng mà còn theo từng phần tử.
```

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 8]
```diff
@@ -276,7 +277,11 @@ That means that after computing the linear portion of the layer, we can calculat
 This is true for most activation functions (the batch normalization operation will be introduced in :numref:`sec_batch_norm` is a notable exception to that rule).
 -->
 
-*dịch đoạn phía trên*
+Khác với các quy ước thông thường, chúng ta định nghĩa hàm phi tuyến $\sigma$ áp dụng cho các giá trị đầu vào theo từng hàng, tức lần lượt từng điểm dữ liệu một.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/mlp_vn.md]: mình thích dịch sát nghĩa hơn
```suggestion
Bằng việc lạm dụng ký hiệu một chút, chúng ta định nghĩa hàm phi tuyến $\sigma$ áp dụng cho các giá trị đầu vào theo từng hàng, tức lần lượt từng điểm dữ liệu một.
```

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 9]
```diff
@@ -276,7 +277,11 @@ That means that after computing the linear portion of the layer, we can calculat
...
+Cần chú ý rằng ta cũng sử dụng quy ước này cho *softmax* để ký hiệu toán tử tính theo từng dòng.
+Thông thường, như trong mục này, các hàm kích hoạt mà chúng ta áp dụng vào tầng ẩn không chỉ đơn thuần theo từng dòng mà còn theo từng phần.
+Điều đó có nghĩa là sau khi tính toán các phần tuyến tính của tầng, chúng ta có thể tính kích hoạt cho từng nút mà không cần quan tâm đến giá trị của các nút còn lại.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
Điều đó có nghĩa là sau khi tính toán các phần tuyến tính của tầng, chúng ta có thể tính giá trị kích hoạt của từng nút mà không cần quan tâm đến giá trị của các nút còn lại.
```

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 10]
```diff
@@ -276,7 +277,11 @@ That means that after computing the linear portion of the layer, we can calculat
...
+Điều này cũng đúng đối với hầu hết các hàm kích hoạt (toán tử chuẩn hoá batch sẽ được giới thiệu trong :numref:`sec_batch_norm` là một trường hợp ngoại lệ của quy tắc này).
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
Điều này cũng đúng đối với hầu hết các hàm kích hoạt (toán tử chuẩn hoá batch được giới thiệu trong :numref:`sec_batch_norm` là một trường hợp ngoại lệ của quy tắc này).
```

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 11]
```diff
@@ -289,15 +294,17 @@ npx.set_np()
...
 
 <!--
 Activation functions decide whether a neuron should be activated or not by calculating the weighted sum and further adding bias with it.
 They are differentiable operators to transform input signals to outputs, while most of them add non-linearity.
 Because activation functions are fundamental to deep learning, let's briefly survey some common activation functions.
 -->
 
-*dịch đoạn phía trên*
+Các hàm kích hoạt quyết định một nơ-ron có được kích hoạt hay không bằng cách tính tổng có trọng số và cộng thêm hệ số điều chỉnh vào nó.
+Chúng là các toán tử khả vi biến đổi các tín hiệu đầu vào thành các tín hiệu đầu ra, mặc dù hầu hết các hàm kích hoạt này đều thêm vào phần phi tuyến tính.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
Chúng là các toán tử khả vi có vai trò biến đổi các tín hiệu đầu vào thành các tín hiệu đầu ra, và hầu hết các hàm kích hoạt này đều thêm sự phi tuyến tính.
```

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 12]
```diff
@@ -289,15 +294,17 @@ npx.set_np()
...
+Bởi vì các hàm kích hoạt rất quan trọng trong học sâu, chúng ta hãy cùng tìm hiểu sơ lược một số hàm kích hoạt thông dụng.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
Bởi vì các hàm kích hoạt rất quan trọng trong học sâu, hãy cùng tìm hiểu sơ lược một số hàm kích hoạt thông dụng.
```

File: chapter_multilayer-perceptrons/mlp_vn.md (hunk 8 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/mlp_vn.md]: okay

File: chapter_multilayer-perceptrons/mlp_vn.md (hunk 6 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/mlp_vn.md]: okay

File: chapter_multilayer-perceptrons/mlp_vn.md (hunk 8 above)
Comment (minhduc0711) [reply chapter_multilayer-perceptrons/mlp_vn.md]: bạn không phải reply mọi comment đâu :))) cái nào thấy ok rùi cứ thầm lặng commit thôi

File: chapter_multilayer-perceptrons/mlp_vn.md (hunk 7 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/mlp_vn.md]: okay

File: chapter_multilayer-perceptrons/mlp_vn.md [hunk 13]
```diff
@@ -596,7 +603,7 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 4 -->
-*
+* Lê Cao Thăng
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/mlp_vn.md]:
```suggestion
* Lê Cao Thăng
* Phạm Minh Đức
```
